### PR TITLE
feat: add policiesSchema.yaml

### PR DESCRIPTION
**Pull request by Ben-Zaad**

### CODE REVIEW

**Review by alexfedin — DISMISSED** [review]

File: pkg/policy/policiesSchema.json [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            "type": "boolean"
+          },
+          "message": {
+            "type": "string"
+          },
+          "rules": {
+            "items": {
+              "type": "object",
+              "properties": {
+                "identifier": {
```
Comment (royhadad) [inline pkg/policy/policiesSchema.json:32]: missing property "messageOnFailure"

File: pkg/policy/policiesSchema.json [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                  "type": "string",
+                  "pattern": "^([A-Z0-9])+(_?[A-Z0-9])*$"
```
Comment (royhadad) [inline pkg/policy/policiesSchema.json]: to the best of my knowledge, we don't have this regex limitation in our backend validator

File: pkg/policy/policiesSchema.json [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+              ]
+            }
+          }
+        }
+      }
+    },
+    "customRules": {
+      "type": "array",
+      "items": {
+        "identifier": "string",
```
Comment (royhadad) [inline pkg/policy/policiesSchema.json]: I think you meant to write:
```
type: object
properties: {
    identifier: "string"
}
```

File: pkg/policy/policiesSchema.json [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        "pattern": "^([A-Z0-9])+(_?[A-Z0-9])*$",
```
Comment (royhadad) [inline pkg/policy/policiesSchema.json]: incorrect regex, this is what we have in backend:
Screen Shot 2022-03-16 at 14 48 04

File: pkg/policy/policiesSchema.json (hunk 3 above)
Comment (royhadad) [inline pkg/policy/policiesSchema.json]: we need to check that identifier is not in default rules, this is from backend:
Screen Shot 2022-03-16 at 14 48 42

File: pkg/policy/policiesSchema.json [hunk 5]
```diff
@@ -0,0 +1,78 @@
+{
+  "type": "object",
+  "properties": {
+    "apiVersion": {
+      "type": "string",
+      "enum": [
+        "v1"
+      ]
+    },
+    "message": {
```
Comment (royhadad) [inline pkg/policy/policiesSchema.json]: what is this key "message"?

File: pkg/policy/policiesSchema.json [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        "identifier": "string",
+        "pattern": "^([A-Z0-9])+(_?[A-Z0-9])*$",
+        "defaultMessageOnFailure": {
+          "type":"string"
+        },
+        "jsonSchema": {
+          "type": "string"
+        },
+        "schema": {
+          "type": "object"
```
Comment (royhadad) [inline pkg/policy/policiesSchema.json:68]: also here, we need to check if it's a valid json schema, in some way

File: pkg/policy/policiesSchema.json [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    "customRules": {
+      "type": "array",
+      "items": {
+        "identifier": "string",
+        "pattern": "^([A-Z0-9])+(_?[A-Z0-9])*$",
+        "defaultMessageOnFailure": {
+          "type":"string"
+        },
+        "jsonSchema": {
+          "type": "string"
```
Comment (royhadad) [inline pkg/policy/policiesSchema.json:65]: We need to check if it's a valid json schema, in some way

File: pkg/policy/policiesSchema.json (hunk 3 above)
Comment (royhadad) [inline pkg/policy/policiesSchema.json]: what about the optional "name" key?

File: pkg/policy/policiesSchema.json [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    "apiVersion": {
+      "type": "string",
+      "enum": [
+        "v1"
+      ]
+    },
+    "message": {
+      "type": "string",
+      "enum": [
+        "Should have exact one policy set as default",
```
Comment (royhadad) [inline pkg/policy/policiesSchema.json]: where are the checks for these 3?
"Should have exact one policy set as default"
"All policy names should be unique"
"All rule identifiers should be unique"

File: pkg/policy/policiesSchema.json (hunk 1 above)
Comment (royhadad) [inline pkg/policy/policiesSchema.json:32]: where is the check that validates rule identifier exists in either custom rules or default rules?

**Review by royhadad — CHANGES_REQUESTED** [review]

File: pkg/policy/policiesSchema.json (hunk 3 above)
Comment (Ben-Zaad) [reply pkg/policy/policiesSchema.json]: I think this is one of the abilities that we are losing due to JSON Schema limitations.

https://github.com/json-schema-org/json-schema-spec/wiki/Scope-of-JSON-Schema-Validation

File: pkg/policy/policiesSchema.json (hunk 1 above)
Comment (Ben-Zaad) [reply pkg/policy/policiesSchema.json:32]: I think this is one of the abilities that we are losing due to JSON Schema limitations.

https://github.com/json-schema-org/json-schema-spec/wiki/Scope-of-JSON-Schema-Validation

**Review by dimabru — APPROVED** [review]

**Review by royhadad — APPROVED** [review]

**Pull request merged**